invenio_vocabularies/contrib/affiliations/datastreams.py

```diff
@@ -96,15 +99,24 @@ def _entry_id(self, entry):
         """Get the id from an entry."""
         return entry["id"]
 
-    def write(self, stream_entry, *args, **kwargs):
-        """Writes the input entry using a given service."""
-        entry = stream_entry.entry
-
-        return super().write(stream_entry, *args, **kwargs)
-
-    def write_many(self, stream_entries, *args, **kwargs):
-        """Writes the input entries using a given service."""
-        return super().write_many(stream_entries, *args, **kwargs)
+    def _do_update(self, entry):
+        vocab_id = self._entry_id(entry)
+        current = self._resolve(vocab_id)
+        updated = deepcopy(current.to_dict())
+
+        if "identifiers" in entry:
+            # For each new identifier
+            for new_identifier in entry["identifiers"]:
+                # Either find an existing identifier with the same scheme and update the "identifier" value
+                for existing_identifier in updated["identifiers"]:
+                    if existing_identifier["scheme"] == new_identifier["scheme"]:
+                        existing_identifier["identifier"] = new_identifier["identifier"]
+                        break
+                # Or add the new identifier to the list of identifiers
+                else:
+                    updated["identifiers"].append(new_identifier)
```

Comment:
We now have an implementation of the update logic very specific to the use case of updating/adding identifiers to a list of identifiers, making sure that we do not overwrite existing identifiers, and making sure that we do not end up with two identifiers for a given scheme.